### PR TITLE
Ignore session ID's shorter than 32 bytes instead of erroring out

### LINKED ISSUE #6408: Compatibility issues with other TLS libraries
### Version

5.5.1 and below

### Description

I think there are two compatibility issues in the TLS 1.2 implementation:

(1) When processing the **SessionID** field in the ClientHello message, the wolfssl server side seems to respond normally only to **SessionID** of length **32** at present.

However, RFC5246 specifies the SessionID length as `opaque SessionID<0..32>`.

There may be compatibility issues with other TLS library clients.

(2) It seems that wolfssl server has **support upper limit 150 for Cipher Suites** in ClientHello.

In general, this upper number is sufficient, but is there a potential compatibility issue with this implementation? 

For example, a client with an individual TLS implementation library will send all the cipher suites it supports, and the number happens to be greater than 150.

### PR DESCRIPTION
**Pull request by julek-wolfssl**
Fixes https://github.com/wolfSSL/wolfssl/issues/6408

### CODE REVIEW

**Review by dgarske — APPROVED** [review]
Over to @SparkiDev . I would like him to review this also

File: src/tls13.c [hunk 1]
```diff
@@ -4100,26 +4144,21 @@ int SendTls13ClientHello(WOLFSSL* ssl)
         args->idx += DTLS_RECORD_EXTRA + DTLS_HANDSHAKE_EXTRA;
 #endif /* WOLFSSL_DTLS13 */
 
-    /* Version | Random | Session Id | Cipher Suites | Compression */
-    args->length = VERSION_SZ + RAN_LEN + ENUM_LEN + suites->suiteSz +
+    /* Version | Random | Cipher Suites | Compression */
+    args->length = VERSION_SZ + RAN_LEN + suites->suiteSz +
             SUITE_LEN + COMP_LEN + ENUM_LEN;
+#if defined(WOLFSSL_TLS13_MIDDLEBOX_COMPAT)
+    ssl->options.tls13MiddleBoxCompat = 1;
```
Comment (SparkiDev) [inline src/tls13.c:4151]: Why are we forcing this here?
It is meant to be something the application chooses to do.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/tls13.c (hunk 1 above)
Comment (julek-wolfssl) [reply src/tls13.c:4151]: That's how it was before my changes. I just preserved the logic with my changes. As to why its forced, I think that when the middlebox macro is on we are supposed to always be acting middlebox compatible.

Comment (SparkiDev) [reply src/tls13.c:4151]: I think the logic was that we are sending the session Id and it is middle box compatibility thing and so we are now doing compatibility.
I don't want it forced on. We have chosen to always send it now regardless of compatibility.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/tls13.c (hunk 1 above)
Comment (julek-wolfssl) [reply src/tls13.c:4151]: This line is only in effect when `WOLFSSL_TLS13_MIDDLEBOX_COMPAT` is defined. This is later overwritten if we are doing quic. This is the same logic as before. Previously, when `WOLFSSL_TLS13_MIDDLEBOX_COMPAT` was defined we would also always force middlebox compatibility. Would you like me to change the behaviour in this PR? I'm not sure if we don't force middlebox compatibility in other locations too. Those would also need to be reviewed.

Comment (SparkiDev) [reply src/tls13.c:4151]: You are refusing to make a requested change.
Fine.
Do another PR with the change and review all other instances.

**Review by SparkiDev — APPROVED** [review]

File: src/tls13.c (hunk 1 above)
Comment (julek-wolfssl) [reply src/tls13.c:4151]: I don't mind making the change. I'm just making sure that we are on the same page about what the requested end result is. I'll tackle this in a new PR.

**Pull request merged**